include/seastar/http/reply.hh

```diff
@@ -44,8 +44,6 @@ namespace seastar {
 
 SEASTAR_MODULE_EXPORT_BEGIN
 
-struct http_response;
```

Comment:
I assume that this "seastar::http_response" type, although is public, should not have been used by any user and we don't need to keep support for it. But I'm not 100% convinced... Can you please convince me?

Reply:
There are no places in http code that export this type anywhere. It's just returned from response parser to websocket and, after http client appeared, to http client

Reply:
i have a FTBFS after bumping up the seastar submodule in scylla by accident, like
```
/home/kefu/dev/scylladb/locator/azure_snitch.cc:88:18: error: no member named '_status_code' in 'seastar::http::reply'
   88 |         if (rsp->_status_code != static_cast<int>(http::reply::status_type::ok)) {
      |             ~~~~~^
```

the caller in scylla looks like
```c++
        auto rsp = parser.get_parsed_response();
        if (rsp->_status_code != static_cast<int>(http::reply::status_type::ok)) {
            throw std::runtime_error(format("Error: HTTP response status {}", rsp->_status_code));
        }
```

@xemul so 

1. we do have a user of this class, 
2. if we want to bump the seastar module, we might want to update the caller sites in `scylladb/locator/*_snitch.cc` accordingly.

i am copying my local change in hope to reduce the workload to create the patch on the scylla's front:
```diff
diff --git a/locator/azure_snitch.cc b/locator/azure_snitch.cc
index 062ec78f83..89b7e5bd82 100644
--- a/locator/azure_snitch.cc
+++ b/locator/azure_snitch.cc
@@ -85,8 +85,8 @@ future<sstring> azure_snitch::azure_api_call(sstring path) {
 
         // Read HTTP response header first
         auto rsp = parser.get_parsed_response();
-        if (rsp->_status_code != static_cast<int>(http::reply::status_type::ok)) {
-            throw std::runtime_error(format("Error: HTTP response status {}", rsp->_status_code));
+        if (rsp->_status != http::reply::status_type::ok) {
+            throw std::runtime_error(format("Error: HTTP response status {}", rsp->_status));
         }
 
         auto it = rsp->_headers.find("Content-Length");
diff --git a/locator/ec2_snitch.cc b/locator/ec2_snitch.cc
index d27a52b866..76123bd28e 100644
--- a/locator/ec2_snitch.cc
+++ b/locator/ec2_snitch.cc
@@ -119,13 +119,13 @@ future<sstring> ec2_snitch::aws_api_call_once(sstring addr, uint16_t port, sstri
 
             // Read HTTP response header first
             auto _rsp = _parser.get_parsed_response();
-            auto rc = _rsp->_status_code;
+            auto rc = _rsp->_status;
             // Verify EC2 instance metadata access
-            if (rc == 403) {
+            if (rc == http::reply::status_type::forbidden) {
                 return make_exception_future<sstring>(std::runtime_error("Error: Unauthorized response received when trying to communicate with instance metadata service."));
             }
-            if (_rsp->_status_code != static_cast<int>(http::reply::status_type::ok)) {
-                return make_exception_future<sstring>(std::runtime_error(format("Error: HTTP response status {}", _rsp->_status_code)));
+            if (_rsp->_status != http::reply::status_type::ok) {
+                return make_exception_future<sstring>(std::runtime_error(format("Error: HTTP response status {}", _rsp->_status)));
             }
 
             auto it = _rsp->_headers.find("Content-Length");
diff --git a/locator/gce_snitch.cc b/locator/gce_snitch.cc
index eff406bbaa..5c052a4d9d 100644
--- a/locator/gce_snitch.cc
+++ b/locator/gce_snitch.cc
@@ -99,8 +99,8 @@ future<sstring> gce_snitch::gce_api_call(sstring addr, sstring cmd) {
 
         // Read HTTP response header first
         auto rsp = parser.get_parsed_response();
-        if (rsp->_status_code != static_cast<int>(http::reply::status_type::ok)) {
-            throw std::runtime_error(format("Error: HTTP response status {}", rsp->_status_code));
+        if (rsp->_status != http::reply::status_type::ok) {
+            throw std::runtime_error(format("Error: HTTP response status {}", rsp->_status));
         }
 
         auto it = rsp->_headers.find("Content-Length");
diff --git a/replica/table.cc b/replica/table.cc
index e1c1d27a86..c2ced7bcc8 100644
--- a/replica/table.cc
+++ b/replica/table.cc
@@ -1503,7 +1503,7 @@ future<> compaction_group::stop() noexcept {
 }
 
 void compaction_group::clear_sstables() {
-    *_main_sstables = _t._compaction_strategy.make_sstable_set(_t._schema);
+    _main_sstables = make_lw_shared<sstables::sstable_set>(_t._compaction_strategy.make_sstable_set(_t._schema));
     _maintenance_sstables = _t.make_maintenance_sstable_set();
 }
 ```

Reply:
I don't remember the details, but it seems that the old API requiring static_cast was indeed crappy so it's good we fixed it, so it's a good idea to send a Scylla patch to update Seastar together with your patch here. Except perhaps the last change which doesn't seem related?

Reply:
In fact Kefu's point is correct -- the locator/ code really used old data type and ~~w~~me missed that fact because of the `auto` camouflage :( The plan was to stop using http response parser directory (scylladb/scylla#11924). Some better excuse is that we sort of decided to stop pushing custom snitch drivers forward, move this whole code into scripts and re-use gossiping-property-file driver (scylladb/scylladb#12306)

Reply:
Can you please either send a patch to revert the disappearance of this type from Seastar, or a patch to Scylla to stop using the API, as you see fit?

In a perfect world, we shouldn't just stop supporting a public type in Seastar (and this is why I asked about this in my review), but in a non-perfect world sometimes an API can be so terribly-designed and so rarely used that it might be "good enough" to withdraw it from Seastar, which isn't exactly a library used by millions.

Reply:
I support the
> sometimes an API can be so terribly-designed and so rarely used that it might be "good enough" to withdraw it from Seastar

and in order to proceed the patch from Kefu (excluding the last hunk) should be merged together with the relevant seastar update. Is it (the seastar update) happening already?

Reply:
I wasn't planning on doing it, but I can do it if you or Kefu aren't planning on it.

Reply:
okay. let me create a patch to bump up the Seastar submodule in scylla and include the proposed change in that change.

Reply:
https://github.com/scylladb/scylladb/pull/14552